### PR TITLE
feat: Adds `Gift` component

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

Depends on https://github.com/vtex/faststore/pull/1443

- Adding `Gift` component to starters
- It's not being used in the application yet.
- Only listed on the [storybook](https://nextjs-store-storybook-git-feat-fs-552-gift-component-faststore.vercel.app/?path=/docs/molecules-gift--default-story).

## How does it work?

It displays an additional item to purchase as a gift.

image

## How to test it?

You can test it in the [storybook preview](https://nextjs-store-storybook-git-feat-fs-552-gift-component-faststore.vercel.app/?path=/story/molecules-gift--default-story)

## References

Gift component on `faststore/ui` https://github.com/vtex/faststore/pull/1443

## Checklist

<em>You may erase this after checking them all :wink:</em>

**PR Title and Commit Messages**
- [x] PR title and commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, and `test`

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Features`, `Bug fixes`, `Chore`, `Documentation`, `Style changes`, `Refactoring`, `Performance`, and `Test`
- [x] Added the component, hook, or path name in-between backticks (\`\`) - *if applicable, e.g., `ComponentName` component, `useWindowDimensions` hook*

**Dependencies**
- [ ] Committed the `yarn.lock` and `bun.lockb` file when there were changes to the packages

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*
- [ ] For documentation changes, ping `@carolinamenezes` or `@PedroAntunesCosta` to review and update

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,10 +31,11 @@
...
     "@envelop/validation-cache": "^2.2.0",
-    "@faststore/api": "^1.11.4",
-    "@faststore/graphql-utils": "^1.10.34",
-    "@faststore/sdk": "^1.10.34",
-    "@faststore/ui": "^1.11.2",
+    "@faststore/api": "https://pkg.csb.dev/vtex/faststore/commit/1c50b583/@faststore/api",
+    "@faststore/graphql-utils": "https://pkg.csb.dev/vtex/faststore/commit/1c50b583/@faststore/graphql-utils",
+    "@faststore/sdk": "https://pkg.csb.dev/vtex/faststore/commit/1c50b583/@faststore/sdk",
+    "@faststore/styles": "https://pkg.csb.dev/vtex/faststore/commit/1c50b583/@faststore/styles",
+    "@faststore/ui": "https://pkg.csb.dev/vtex/faststore/commit/1c50b583/@faststore/ui",
```
Comment (hellofanny) [inline package.json:38]: Update it after merging https://github.com/vtex/faststore/pull/1443 🎗️

File: src/components/ui/Gift/gift.module.scss [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  // Design Tokens for Gift
+  // --------------------------------------------------------
+
+  // Default properties
+
+  --fs-gift-height                         : var(--fs-spacing-13);
+  --fs-gift-gap                            : var(--fs-spacing-2);
+
+  --fs-gift-border-width                   : var(--fs-border-width);
+  --fs-gift-border-color                   : var(--fs-border-color-light-active);
```
Comment (hellofanny) [inline src/components/ui/Gift/gift.module.scss:14]: @renatamottam shouldn't this color be the same as `CartItem` border? `--fs-border-color-light`? But in the Figma the color specified was this one 🤔

File: src/components/ui/Gift/gift.module.scss [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  [data-fs-gift-content] {
+    display: grid;
+    grid-column: 2 / span 4;
+    padding: var(--fs-gift-content-padding);
+  }
+
+  [data-gift-product-title] {
+    margin-bottom: var(--fs-gift-margin-bottom);
```
Comment (renatamottam) [inline src/components/ui/Gift/gift.module.scss:67]: Changed to `row-gap`
```suggestion
```

File: src/components/ui/Gift/Gift.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  <TokenRow
+    token="--fs-gift-content-padding"
+    value="var(--fs-gift-gap) var(--fs-gift-gap) var(--fs-gift-gap) 0"
+  />
+</TokenTable>
+
+### Title
+
+<TokenTable>
+  <TokenRow token="--fs-gift-title-margin-bottom" value="var(--fs-spacing-0)" />
```
Comment (renatamottam) [inline src/components/ui/Gift/Gift.stories.mdx:74]: Changed to `row-gap`
```suggestion
```

File: src/components/ui/Gift/gift.module.scss [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  --fs-gift-border-color                   : var(--fs-border-color-light-active);
+  --fs-gift-border-radius                  : var(--fs-border-radius);
+
+  // Content
+  --fs-gift-content-padding                : var(--fs-gift-gap) var(--fs-gift-gap) var(--fs-gift-gap) 0;
+
+  // Title
+  --fs-gift-title-margin-bottom            : var(--fs-spacing-0);
+  --fs-gift-title-line-height              : 1.25;
+  --fs-gift-title-text-size                : var(--fs-text-size-body);
```
Comment (renatamottam) [inline src/components/ui/Gift/gift.module.scss:23]:
```suggestion
  --fs-gift-title-size                     : var(--fs-text-size-body);
```

File: src/components/ui/Gift/gift.module.scss [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+  [data-fs-gift-content] {
+    display: grid;
+    grid-column: 2 / span 4;
+    padding: var(--fs-gift-content-padding);
+  }
+
+  [data-gift-product-title] {
+    margin-bottom: var(--fs-gift-margin-bottom);
+    font-size: var(--fs-gift-title-text-size);
```
Comment (renatamottam) [inline src/components/ui/Gift/gift.module.scss:68]:
```suggestion
    font-size: var(--fs-gift-title-size);
```

File: src/components/ui/Gift/Gift.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  <TokenRow token="--fs-gift-title-line-height" value="1.25" />
+  <TokenRow
+    token="--fs-gift-title-text-size"
```
Comment (renatamottam) [inline src/components/ui/Gift/Gift.stories.mdx:77]:
```suggestion
    token="--fs-gift-title-size"
```

**Review by renatamottam — COMMENTED** [review]
Nice work! Left a couple improvement suggestions :)

File: src/components/ui/Gift/Gift.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  />
+</TokenTable>
+
+### Icon
+
+<TokenTable>
+  <TokenRow token="--fs-gift-icon-size" value="1.75rem" />
+  <TokenRow token="--fs-gift-icon-padding" value="var(--fs-spacing-0)" />
+  <TokenRow token="--fs-gift-icon-color" value="var(--fs-gift-title-color)" />
+  <TokenRow token="--fs-gift-icon-bkg-color" value="var(--fs-gift-body-bkg)" />
```
Comment (renatamottam) [inline src/components/ui/Gift/Gift.stories.mdx:90]: I think the `--fs-gift-body-bkg` token is missing both on `module.scss` and `TokenTable`

```suggestion
  <TokenRow
    token="--fs-gift-icon-color"
    value="var(--fs-gift-title-color)"
    globalValue="var(--fs-color-text)"
    isColor
  />
  <TokenRow
    token="--fs-gift-icon-bkg-color"
    value="var(--fs-gift-body-bkg)"
    globalValue="var(--fs-color-body-bkg)"
    isColor
  />
```

**Review by renatamottam — APPROVED** [review]
Great work, fan! Left a tiny fix but it's approved after that :)

File: src/components/ui/Gift/Gift.stories.mdx (hunk 8 above)
Comment (hellofanny) [reply src/components/ui/Gift/Gift.stories.mdx:90]: My bad! Gonna fix it.

File: yarn.lock [hunk 9]
```diff
@@ -3,11 +3,12 @@
 
 
 "@ampproject/remapping@^2.1.0":
-  version "2.1.2"
-  resolved "https://registry.yarnpkg.com/@ampproject/remapping/-/remapping-2.1.2.tgz#4edca94973ded9630d20101cd8559cedb8d8bd34"
-  integrity sha512-hoyByceqwKirw7w3Z7gnIIZC3Wx3J484Y3L/cMpXFbr7d9ZQj2mODrirNzcJa+SM3UlpWXYvKV4RlRpFXlWgXg==
+  version "2.2.0"
```
Comment (filipewl) [inline yarn.lock:6]: I'm finding most of the modifications in this file strange. I was assuming that only lines related to the `@faststore` packages bump would be touched like we had this last time [here](https://github.com/vtex-sites/nextjs.store/pull/187/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deL1436-R1461). Can you double check these changes are needed?

You can try to revert this whole file to the `main`'s version using `git checkout main -- yarn.lock` then run `yarn install` again.

**Review by filipewl — COMMENTED** [review]
Looks good, but left a comment about the `yarn.lock` changes.

Also, since the `package.json` has been modified, remember to also run `bun install` and commit the `bun.lockb` file.

File: yarn.lock (hunk 9 above)
Comment (hellofanny) [reply yarn.lock:6]: okay! Whenever I try to update this file, the modifications look odd. Lemme try again!

**Comment by hellofanny** [comment]
closing this PR in favor of https://github.com/vtex-sites/nextjs.store/pull/234

**Pull request closed without merging**